web-common/src/main/java/org/openmrs/eip/web/contoller/BaseRestController.java

```diff
@@ -52,6 +53,39 @@ public Object doGet(Long id) {
 		    "jpa:" + getName() + "?query=SELECT c FROM " + getName() + " c WHERE c.id = " + id, null, getClazz());
 	}
 	
+	public Map<String, Object> doSearchByPeriod(String startDate, String endDate) {
```

Comment:
In fact this method is not reusable across different entities, this search is specific to sender sync archives and you should search against dateSent and not dateCreated

Reply:
What does the requirement say?

Reply:
No move it to the subclass where you are call it

Reply:
If you want to keep it here and make it reusable then you need to give it the date field property name to search against as an argument

Reply:
Done. I added a third argument to indicate the property that will be based on the search